src/interfaces/IMulticall3.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2 <0.9.0;
+
+pragma experimental ABIEncoderV2;
+
+interface IMulticall3 {
+    struct Call {
+        address target;
+        bytes callData;
+    }
+
+    struct Call3 {
+        address target;
+        bool allowFailure;
+        bytes callData;
+    }
+
+    struct Call3Value {
+        address target;
+        bool allowFailure;
+        uint256 value;
+        bytes callData;
+    }
+
+    struct Result {
+        bool success;
+        bytes returnData;
+    }
+
+    function aggregate(Call[] calldata calls)
+        external
+        payable
+        returns (uint256 blockNumber, bytes[] memory returnData);
+
+    function aggregate3(Call3[] calldata calls) external payable returns (Result[] memory returnData);
+
+    function aggregate3Value(Call3Value[] calldata calls) external payable returns (Result[] memory returnData);
+
+    function blockAndAggregate(Call[] calldata calls)
+        external
+        payable
+        returns (uint256 blockNumber, bytes32 blockHash, Result[] memory returnData);
+
+    function getBasefee() external view returns (uint256 basefee);
+
+    function getBlockHash(uint256 blockNumber) external view returns (bytes32 blockHash);
+
+    function getBlockNumber() external view returns (uint256 blockNumber);
+
+    function getChainId() external view returns (uint256 chainid);
+
+    function getCurrentBlockCoinbase() external view returns (address coinbase);
+
+    function getCurrentBlockDifficulty() external view returns (uint256 difficulty);
```

Comment:
this will return 00000.... post merge, why not replace with prevrando?

Reply:
Yes its an alias, either way it should be changed or add prevrando if support for PoW chains is wanted I suppose.

👍

Reply:
Actually DIFFICULTY will return zeroed out response in the block im pretty sure

Reply:
Given that the contract is already deployed on 50+ chains, I don't think this is worth changing. Users just need to understand how it behaves differently across different chains. Similarly, there is a `getBasefee` method that will revert on chains that don't support EIP-1559. Perhaps a PR to the README to document these differences would be valuable

Reply:
yea did not realize this was your deployed contract 🙃 

💯

Reply:
Created https://github.com/mds1/multicall/issues/57 to track the documentation changes, thanks 🙂

Going to mark this comment resolved